dev/release/03-source.sh

```diff
@@ -63,18 +55,16 @@ main() {
 
     # commit to svn
     svn add "tmp/${tag}"
-    svn ci -m "Apache Arrow ADBC ${version} RC${rc_number}" "tmp/${tag}"
+    if [[ "${DRY_RUN}" = 0 ]]; then
```

Comment:
Bash uses `-eq` for number comparison. 

```suggestion
    if [[ ${DRY_RUN} -eq 0 ]]; then
```